pkg/build/api/validation/validation.go

```diff
@@ -144,23 +144,22 @@ func validateBuildConfigOutput(output *buildapi.BuildOutput) errs.ValidationErro
 func validateStrategy(strategy *buildapi.BuildStrategy) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
 
-	if len(strategy.Type) == 0 {
+	switch {
+	case len(strategy.Type) == 0:
```

Comment:
Wouldn't it be better to expose something like `NoStrategy` from the buildapi (eg. `const NoStrategy = ""`) and keep the switch as it was before here?